src/components/FacetHeader.tsx

```diff
@@ -0,0 +1,89 @@
+import '@isrd-isi-edu/chaise/src/assets/scss/_faceting.scss';
+
+import { useRef, useState } from 'react';
+
+// Components
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
+import Tooltip from 'react-bootstrap/Tooltip';
+import DisplayValue from '@isrd-isi-edu/chaise/src/components/display-value';
+
+type FacetHeaderProps = {
+  key: string;
+  headerText: any;
+  showTooltipIcon: any;
+  tooltipContent: any;
+};
+
+/**
+ * @param key key prop to uniquely identify panel header item
+ * @param headerText headerText prop that holds content to show as header
+ * @param showTooltipIcon prop to enable tooltip for header text
+ * @param tooltipContent text to be shown on hover of header text
+ * @returns FacetHeader Component
+ */
+const FacetHeader = ({
+  key,
+  headerText,
+  showTooltipIcon,
+  tooltipContent,
+}: FacetHeaderProps) => {
+  /**
+   * @contentRef variable to store ref of facet header text
+   * @show state variable to control whether to show tooltip or not
+   */
+  const contentRef = useRef(null);
+  const [show, setShow] = useState(false);
+
+  // Function to check the text overflow.
+  const isTextOverflow = (element) => {
+    if (element) {
+      return element.offsetWidth < element.scrollWidth;
+    }
+    return false;
+  };
+
+  return (
+    <OverlayTrigger
+      trigger='hover'
+      placement='right'
+      overlay={
+        <Tooltip style={{ maxWidth: '50%', whiteSpace: 'nowrap' }}>
+          {contentRef &&
```

Comment:
Given that it has too many different conditions now, move this to a function and use proper if statements. This would help with readability of your code.

Reply:
Made a separate function.